lib/codegen/fromcto/vocabulary/vocabularyvisitor.js

```diff
@@ -54,6 +54,10 @@ class VocabularyVisitor {
             return this.visitProperty(thing, parameters);
         } else if (thing.isEnumValue?.()) {
             return this.visitProperty(thing, parameters);
+        } else if (thing.isKey?.()) {
+            return this.visitMapKey(thing, parameters);
+        } else if (thing.isValue?.()) {
```

Comment:
I guess, it's a bit late to do anything about it now, but `isValue` looks a lot more generic than it should be. We could have made it like `isMapValue`, just as we did for declarations. Just a remark, nothing actionable.

Reply:
Maybe to future-proof this, if possible, we can check if the parent of the `thing` is a map, os something similar to scope this to map declarations only. The same goes for the keys.

Reply:
For now I believe keys and values are only scoped to Maps as of now.

Reply:
If you are suggesting for readability and interpretation, that's why kept the visitor method names as `visitMapKey` `visitMapValue`.
I agree with you the method name is very generic.